yarn test failing because 127.0.0.1:7999 is already in use

I get the failure shown in the text below even in a totally clean and up to date `master`. So I visited the socket address mentioned in my browser and saw the image below. As far as I can tell, Pledger itself is not running.

![Screen Shot 2021-03-04 at 2 40 43 PM](https://user-images.githubusercontent.com/273868/110040778-2d610a00-7cf8-11eb-8589-4b2897deba68.png)



```
1):cosmic-swingset(master)$ yarn test test/test-home.js 
yarn run v1.22.5
$ ava test/test-home.js

# connecting
.
o
{
  code: 'EADDRINUSE',
  errno: -48,
  syscall: 'listen',
  address: '::',
  port: 9464
}
make: *** [scenario3-run] Error 1
rm -rf t3
bin/ag-solo init t3 --egresses=fake --webport=7999
2021-03-04T22:36:49.937Z ag-solo: init: ag-solo initialized in t3
2021-03-04T22:36:49.939Z ag-solo: init: HTTP/WebSocket will listen on 127.0.0.1:7999
Execute `make scenario3-run` to run the client and simulated chain
(cd t3 && \
                        ../bin/ag-solo set-fake-chain --delay=0 mySimGCI)
cd t3 && OTEL_EXPORTER_PROMETHEUS_PORT=9464 ../bin/ag-solo start
2021-03-04T22:37:01.272Z start: adding HTTP/WS listener on 127.0.0.1:7999
2021-03-04T22:37:01.277Z web: Serving static files from /Users/markmiller/src/ongithub/agoric/agoric-sdk/packages/cosmic-swingset/t3/html
2021-03-04T22:37:01.282Z start: adding follower/sender for fake chain mySimGCI
2021-03-04T22:37:01.287Z launch-chain: Launching SwingSet kernel

  setup

  Rejected promise returned by test. Reason:

  2



  teardown

  Rejected promise returned by test. Reason:

  Error {
    code: 'ESRCH',
    errno: -3,
    syscall: 'kill',
    message: 'kill ESRCH',
  }

  › kill (packages/cosmic-swingset/test/captp-fixture.js:125:13)
  › packages/cosmic-swingset/test/test-home.js:179:9



  Unhandled rejection in test/test-home.js

  Error: "test fixture" connection closed
    at disconnectReason (packages/captp/lib/captp.js:48:5)
    at CTP_DISCONNECT (packages/captp/lib/captp.js:348:24)
    at dispatch (packages/captp/lib/captp.js:388:9)
    at abort (packages/captp/lib/captp.js:400:5)
    at WebSocket.<anonymous> (packages/cosmic-swingset/test/captp-fixture.js:97:11)
    at WebSocket.emit (events.js:315:20)
    at WebSocket.EventEmitter.emit (domain.js:467:12)
    at Socket.emit (events.js:315:20)
    at Socket.EventEmitter.emit (domain.js:467:12)
    at TCP.<anonymous> (net.js:673:12)

  › disconnectReason (packages/captp/lib/captp.js:48:5)
  › CTP_DISCONNECT (packages/captp/lib/captp.js:348:24)
  › dispatch (packages/captp/lib/captp.js:388:9)
  › abort (packages/captp/lib/captp.js:400:5)
  › WebSocket.<anonymous> (packages/cosmic-swingset/test/captp-fixture.js:97:11)

  ─

  2 hooks failed
  1 unhandled rejection
error Command failed with exit code 1.
```

--- a/packages/cosmic-swingset/test/captp-fixture.js
+++ b/packages/cosmic-swingset/test/captp-fixture.js
@@ -10,29 +10,35 @@ const PORT = 7999;
 export { E };
 
 export async function makeFixture() {
+  const noisy = process.env.NOISY;
   const accessToken = await getAccessToken(PORT);
 
   let expectedToExit = false;
   let buf = '';
+  const stdio = noisy
+    ? ['ignore', 'inherit', 'inherit']
+    : ['ignore', 'pipe', 'pipe'];
   const cp = spawn(
     'make',
     ['scenario3-setup', 'scenario3-run', `BASE_PORT=${PORT}`],
     {
       cwd: `${__dirname}/..`,
       env: { ...process.env, PORT },
-      stdio: ['ignore', 'pipe', 'pipe'],
+      stdio,
       detached: true,
     },
   );
 
-  cp.stdout.on('data', chunk => (buf += chunk.toString('utf-8')));
-  cp.stderr.on('data', chunk => {
-    const msg = chunk.toString('utf-8');
-    if (!msg.match(/^make: \*\*\*.*99/)) {
-      // Write chunks that don't describe the exit status.
-      process.stderr.write(chunk);
-    }
-  });
+  if (!noisy) {
+    cp.stdout.on('data', chunk => (buf += chunk.toString('utf-8')));
+    cp.stderr.on('data', chunk => {
+      const msg = chunk.toString('utf-8');
+      if (!msg.match(/^make: \*\*\*.*99/)) {
+        // Write chunks that don't describe the exit status.
+        process.stderr.write(chunk);
+      }
+    });
+  }
 
   /** @type {WebSocket} */
   let ws;
@@ -122,10 +128,11 @@ export async function makeFixture() {
     // Don't kill on exit anymore, as we're doing it now.
     process.off('exit', kill);
     // console.log('killing!');
-    process.kill(-cp.pid, 'SIGINT');
+    process.kill(-cp.pid, 'SIGTERM');
   }
 
   process.on('exit', kill);
   process.on('SIGINT', kill);
+  process.on('SIGTERM', kill);
   return { homeP: connect(), kill };
 }

--- a/packages/cosmic-swingset/test/captp-fixture.js
+++ b/packages/cosmic-swingset/test/captp-fixture.js
@@ -9,8 +9,7 @@ const PORT = 7999;
 // Ensure we're all using the same HandledPromise.
 export { E };
 
-export async function makeFixture() {
-  const noisy = process.env.NOISY;
+export async function makeFixture(noisy = false) {
   const accessToken = await getAccessToken(PORT);
 
   let expectedToExit = false;

--- a/packages/cosmic-swingset/test/test-home.js
+++ b/packages/cosmic-swingset/test/test-home.js
@@ -8,7 +8,7 @@ import { makeFixture, E } from './captp-fixture';
 let home;
 let teardown;
 test.before('setup', async t => {
-  const { homeP, kill } = await makeFixture();
+  const { homeP, kill } = await makeFixture(process.env.NOISY);
   teardown = kill;
   home = await homeP;
   t.truthy('ready');

